### PR TITLE
Fix `_free_weak_ref` error

### LINKED ISSUE #74016: AttributeError: 'NoneType' object has no attribute '_free_weak_ref'
### 🐛 Describe the bug

https://github.com/pytorch/pytorch/pull/66970#issuecomment-1063643284 produces these warnings

```
(/home/ezyang/local/pytorch-tmp-env) [ezyang@devvm066.ash0 ~/local/labs] MASTER_ADDR=localhost MASTER_PORT=29500 python worker0.py  
tensor([-0.6383,  0.2935,  0.6883, -0.9899], requires_grad=True)
Exception ignored in: <function StorageWeakRef.__del__ at 0x7f15538914c0>
Traceback (most recent call last):
  File "/data/users/ezyang/pytorch-tmp/torch/multiprocessing/reductions.py", line 36, in __del__
  File "/data/users/ezyang/pytorch-tmp/torch/storage.py", line 520, in _free_weak_ref
AttributeError: 'NoneType' object has no attribute '_free_weak_ref'
Exception ignored in: <function StorageWeakRef.__del__ at 0x7f15538914c0>
Traceback (most recent call last):
  File "/data/users/ezyang/pytorch-tmp/torch/multiprocessing/reductions.py", line 36, in __del__
  File "/data/users/ezyang/pytorch-tmp/torch/storage.py", line 520, in _free_weak_ref
AttributeError: 'NoneType' object has no attribute '_free_weak_ref'
(/home/ezyang/local/pytorch-tmp-env) [ezyang@devvm066.ash0 ~/local/labs] echo $?
0
```

this shouldn't happen. Investigate.

### Versions

master

cc @ezyang @bhosmer @smessmer @ljk53 @bdhirsh

### PR DESCRIPTION
**Pull request by kurtamohler**
Fixes #74016

### CODE REVIEW

File: test/distributed/rpc/test_share_memory.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import torch
```
Comment (kurtamohler) [inline test/distributed/rpc/test_share_memory.py:1]: I based this test file on https://github.com/pytorch/pytorch/pull/66970#issuecomment-1063643284

Maybe there's a simpler way to test this, but I'm not sure how

**Review by ezyang — APPROVED** [review]

**Comment by kurtamohler** [comment]
Hi @lucysumi, yes, this PR is the solution. Hopefully it will be picked up for the next PyTorch release 1.12. If not, it will be in the release after that, 1.13

**Comment by alinasr-esf** [comment]
hey guys i have same error and try to downgrade torch version but not working 
error change to  "GeForce RTX 3090 with CUDA capability sm_86 is not compatible with the current PyTorch installation"
so now i installed cuda 11.6 with nvidia driver version 510.73.03 and torch version 1.12 but error is still on
has anyone found the solution for this error _free_weak_ref

**Comment by alinasr-esf** [comment]
sorry my torch version is 1.11 and torchvision version is 0.12.0

**Pull request closed without merging**